### PR TITLE
test: make tests work on Windows and in GCE

### PR DESCRIPTION
**Pull request by alexander-fenster**
This PR applies the fix from https://github.com/googleapis/synthtool/pull/240 to this repo.

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by alexander-fenster** [comment]
Oh wow. 6 Windows tests are failing for real! I'll take a look and fix in a separate PR.

**Comment by alexander-fenster** [comment]
(This one still can be merged, as it allows Windows builds to run on the right version of Node.js)

File: .kokoro/test.bat [hunk 1]
```diff
@@ -17,7 +17,15 @@
 cd /d %~dp0
 cd ..
 
-call npm install -g npm@latest || goto :error
+@rem The image we're currently running has a broken version of Node.js enabled
+@rem by nvm (v10.15.3), which has no npm bin. This hack uses the functional
+@rem Node v8.9.1 to install npm@latest, it then uses this version of npm to
```
Comment (bcoe) [inline .kokoro/test.bat]: I will try to work with @jkwlui to get this patched first thing next week; hate that we have to install npm twice.

**Comment by alexander-fenster** [comment]
@bcoe @JustinBeckwith Can you please review the test changes? Looks like everything worked.

File: test/test.googleauth.ts [hunk 2]
```diff
@@ -1221,6 +1264,10 @@ describe('googleauth', () => {
   it('should make the request', async () => {
     const url = 'http://example.com';
     const {auth, scopes} = mockGCE();
+
+    // tslint:disable-next-line no-any
+    sinon.stub(auth as any, 'getDefaultServiceProjectId').resolves();
```
Comment (JustinBeckwith) [inline test/test.googleauth.ts]: This should not be required.  This is exactly what `createGetProjectIdNock` is supposed to do.

File: test/test.googleauth.ts [hunk 3]
```diff
@@ -402,18 +403,30 @@ describe('googleauth', () => {
...
+      // git does not create symlinks on Windows
+      return;
+    }
     await auth._getApplicationCredentialsFromFilePath(
       './test/fixtures/goodlink'
     );
   });
 
   it('getApplicationCredentialsFromFilePath should error on invalid symlink', async () => {
+    if (onWindows()) {
```
Comment (JustinBeckwith) [inline test/test.googleauth.ts]: I mean, if windows doesn't create the symlink, it shouldn't matter right?

File: test/test.googleauth.ts [hunk 4]
```diff
@@ -810,6 +823,9 @@ describe('googleauth', () => {
   });
 
   it('getApplicationDefault should cache the credential when using GCE', async () => {
+    // tslint:disable-next-line no-any
+    sinon.stub(auth as any, 'getDefaultServiceProjectId').resolves();
```
Comment (JustinBeckwith) [inline test/test.googleauth.ts]: This is what `createGetProjectIdNock` is supposed to do.  Something is very wrong if we need this

Comment (JustinBeckwith) [reply test/test.googleauth.ts]: (here and below)

File: test/test.googleauth.ts (hunk 3 above)
Comment (alexander-fenster) [reply test/test.googleauth.ts]: Right! Removed this

File: test/test.googleauth.ts (hunk 4 above)
Comment (alexander-fenster) [reply test/test.googleauth.ts]: Changed - please take a look!

File: test/test.googleauth.ts (hunk 2 above)
Comment (alexander-fenster) [reply test/test.googleauth.ts]: Changed - please take a look!

**Comment by alexander-fenster** [comment]
It works! @JustinBeckwith can I get one more look here? Thank you!

File: test/test.googleauth.ts [hunk 5]
```diff
@@ -117,6 +118,8 @@ describe('googleauth', () => {
       .withArgs(wellKnownPathWindows)
       .returnsArg(0);
 
+    sandbox.stub(child_process, 'exec').callsArgWith(1, null, '', null);
```
Comment (JustinBeckwith) [inline test/test.googleauth.ts]: Can we scope this to only mock it with known parameters?  I just want to make sure we're being super explicit about what we want to mock.

```js
sandbox.stub(child_process, 'exec')
  .callThrough()
  .withArgs('gcloud config config-helper --format json')
  .callsArgWith(1, null, '');
```

File: test/test.googleauth.ts [hunk 6]
```diff
@@ -641,12 +654,12 @@ describe('googleauth', () => {
   });
 
   it('getProjectId should return a new projectId the first time and a cached projectId the second time', async () => {
-    mockEnvVar('GCLOUD_PROJECT', fixedProjectId);
+    mockEnvVar('GCLOUD_PROJECT', STUB_PROJECT);
```
Comment (JustinBeckwith) [inline test/test.googleauth.ts]: Where is `STUB_PROJECT` even defined?  What was wrong with just `fixedProjectId`?

File: test/util.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export function onWindows(): boolean {
```
Comment (JustinBeckwith) [inline test/util.ts]: Can we just add this as an `const isWindows = process.platform === 'win32';` at the top of the file?  No need to have this separate for now, and this file would need a lic header if we did

File: test/test.googleauth.ts (hunk 6 above)
Comment (alexander-fenster) [reply test/test.googleauth.ts]: In the same file:

https://github.com/googleapis/google-auth-library-nodejs/blob/47905529d225ca76c020f2400a57fae6150e9536/test/test.googleauth.ts#L42-53

It just happened that we had two variables with same values and same meaning.

File: test/util.ts (hunk 7 above)
Comment (alexander-fenster) [reply test/util.ts]: Sure! I was not sure if we needed this in more than one file when I did that. Changing.

File: test/test.googleauth.ts (hunk 5 above)
Comment (alexander-fenster) [reply test/test.googleauth.ts]: Done! `sinon` has some weird issues with types (it was forcing me to pass at least 2 parameters to `.withArgs` in this case), so I added this `undefined`. By looking at `.d.ts` I couldn't understand what went wrong.

**Comment by JustinBeckwith** [comment]
LGTM after Windows build is passing :)

**Pull request merged**